src/components/connect.js

```diff
@@ -223,20 +223,20 @@ function connect(mapPropsToRequestsToProps, defaults, options) {
 
       refetchDataFromMappings(mappings) {
         mappings = coerceMappings(mappings)
-        Object.keys(mappings).forEach(prop => {
+        return Promise.all(Object.keys(mappings).map(prop => {
           const mapping = mappings[prop]
 
           if (Function.prototype.isPrototypeOf(mapping)) {
             this.setAtomicState(prop, new Date(), mapping, (...args) => {
```

Comment:
I think this one doesn't need a `return`because it's a functional mapping, there's no useful return value for a functional mapping (I guess), so I just placed the return below and it returns `undefined`